runtime/vm/exceptiondescribe.c

```diff
@@ -138,78 +134,68 @@ printStackTraceEntry(J9VMThread * vmThread, void * voidUserData, UDATA bytecodeO
+								vmThread, module->version, J9_STR_NULL_TERMINATE_RESULT, "", 0, versionBuf, J9VM_PACKAGE_NAME_BUFFER_LENGTH, NULL);
+							if (versionBuf != moduleVersionUTF) {
+								freeModuleVersion = TRUE;
+							}
+						}
+					} else {
+						moduleNameUTF = JAVA_BASE_MODULE;
 					}
 				}
+			} else {
+				/* Assume bootstrap code within java.base when the module itself hasn't been created yet  */
+				moduleNameUTF = JAVA_BASE_MODULE;
 			}
 		}
-		if (romMethod->modifiers & J9AccNative) {
+		if (J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccNative)) {
 			sourceFileName = "NativeMethod";
 			sourceFileNameLen = LITERAL_STRLEN("NativeMethod");
-		} else if (sourceFile) {
+		} else if (NULL != sourceFile) {
 			sourceFileName = (char *)J9UTF8_DATA(sourceFile);
 			sourceFileNameLen = (UDATA)J9UTF8_LENGTH(sourceFile);
 		} else {
 			sourceFileName = "Unknown Source";
 			sourceFileNameLen = LITERAL_STRLEN("Unknown Source");
 		}
 		if (NULL != moduleNameUTF) {
-			if (NULL != moduleVersionUTF) {
-				if (lineNumber) {
-					format = j9nls_lookup_message(
-						J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
-						J9NLS_VM_STACK_TRACE_WITH_MODULE_VERSION_LINE,
-						"\tat %.*s.%.*s (%s@%s/%.*s:%u)\n");
-				} else {
-					format = j9nls_lookup_message(
-						J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
-						J9NLS_VM_STACK_TRACE_WITH_MODULE_VERSION,
-						"\tat %.*s.%.*s (%s@%s/%.*s)\n");
-				}
-				j9tty_err_printf(PORTLIB, (char*)format,
-					(UDATA)J9UTF8_LENGTH(className), J9UTF8_DATA(className),
-					(UDATA)J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName),
-					moduleNameUTF, moduleVersionUTF,
-					sourceFileNameLen, sourceFileName,
-					lineNumber); /* line number will be ignored in if it's not used in the format string */
-				if (TRUE == freeModuleVersion) {
-					j9mem_free_memory(moduleVersionUTF);
-				}
+			if (0 != lineNumber) {
+				format = j9nls_lookup_message(
+					J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
+					J9NLS_VM_STACK_TRACE_WITH_MODULE_VERSION_LINE,
+					"\tat %.*s.%.*s (%s@%s/%.*s:%u)\n");
 			} else {
-				if (lineNumber) {
-					format = j9nls_lookup_message(
-						J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
-						J9NLS_VM_STACK_TRACE_WITH_MODULE_LINE,
-						"\tat %.*s.%.*s (%s/%.*s:%u)\n");
-				} else {
-					format = j9nls_lookup_message(
-						J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
-						J9NLS_VM_STACK_TRACE_WITH_MODULE,
-						"\tat %.*s.%.*s (%s/%.*s)\n");
-				}
-				j9tty_err_printf(PORTLIB, (char*)format,
-					(UDATA)J9UTF8_LENGTH(className), J9UTF8_DATA(className),
-					(UDATA)J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName),
-					moduleNameUTF,
-					sourceFileNameLen, sourceFileName,
-					lineNumber); /* line number will be ignored in if it's not used in the format string */
+				format = j9nls_lookup_message(
+					J9NLS_DO_NOT_PRINT_MESSAGE_TAG,
+					J9NLS_VM_STACK_TRACE_WITH_MODULE_VERSION,
+					"\tat %.*s.%.*s (%s@%s/%.*s)\n");
+			}
+			if (NULL == moduleVersionUTF) {
+				moduleVersionUTF = JAVA_SPEC_VERSION_STRING;
```

Comment:
There are two places that set `moduleNameUTF = JAVA_BASE_MODULE;` - can setting `moduleVersionUTF` be done in the same places and the null check be removed?

Reply:
Agreed. Also derive `java.base` module version from `module->version`, if the result is `null`, fall back to `JAVA_SPEC_VERSION_STRING`.